backend/kernelCI_app/views/hardwareDetailsCommitHistoryView.py

```diff
@@ -0,0 +1,201 @@
+from collections import defaultdict
+from django.db import connection
+import json
+from typing import Dict, Literal, List, TypedDict
+from kernelCI_app.helpers.errorHandling import create_error_response
+from kernelCI_app.helpers.logger import log_message
+from django.utils.decorators import method_decorator
+from django.views import View
+from datetime import datetime, timezone
+from django.http import JsonResponse
+from django.views.decorators.csrf import csrf_exempt
+from kernelCI_app.helpers.trees import make_tree_identifier_key
+from kernelCI_app.typeModels.hardwareDetails import (
+    CommitHead,
+    CommitHistoryPostBody,
+    CommitHistoryValidCheckout,
+)
+from pydantic import ValidationError
+
+DEFAULT_DAYS_INTERVAL = 3
+SELECTED_HEAD_TREE_VALUE = "head"
+STATUS_FAILED_VALUE = "FAIL"
```

Comment:
nit: since this is being used in more than one file, it would be nice to have these constants in a third file (maybe `kernelCI_app/constants/hardware.py`) to be properly reused